### ISSUE TITLE
Calls to Office.context.mailbox.item.getSharedPropertiesAsync returns successfully but the targetRestUrl is absent

### ISSUE THREAD

**Issue by davyarris** [OfficeDev/office-js#1240]
## Expected Behavior
[**Office.context.mailbox.item.getSharedPropertiesAsync**](https://docs.microsoft.com/en-us/javascript/api/outlook/office.appointmentcompose?view=outlook-js-1.8#getsharedpropertiesasync-options--callback-) method returns an instance of [**Office.SharedProperties**](https://docs.microsoft.com/en-us/javascript/api/outlook/office.sharedproperties?view=outlook-js-1.8) that contains a valid targetRestUrl.

## Current Behavior
[**Office.context.mailbox.item.getSharedPropertiesAsync**](https://docs.microsoft.com/en-us/javascript/api/outlook/office.appointmentcompose?view=outlook-js-1.8#getsharedpropertiesasync-options--callback-) method returns an instance of [**Office.SharedProperties**](https://docs.microsoft.com/en-us/javascript/api/outlook/office.sharedproperties?view=outlook-js-1.8) **without** the targetRestUrl property.

The user has delegated permissions. The selected message resides in a shared inbox. The addin manifest has the **SupportsSharedFolders** set correctly.

### Steps to Reproduce

1. Create addin manifest, that will allow usage in shared mailboxes with properly set **SupportsSharedFolders** in it.
2. Deploy the addin to https://outlook.office.com/
3. Open the addin for a message in a shared mailbox for **_which the user has delegated permissions_**
4. Execute request to Office.context.mailbox.item.getSharedPropertiesAsync to obtain data Office.SharedProperties for the currently viewed mail
5. No targetRestUrl is found on the SharedProperties object.

```
Office.context.mailbox.getCallbackTokenAsync({
    isRest: true
}, function (result) {
    var id = Office.context.mailbox.convertToRestId(Office.context.mailbox.item.itemId, Office.MailboxEnums.RestVersion.v2_0);
    if (result.status === "succeeded") {
        Office.context.mailbox.item.getSharedPropertiesAsync(function (options) {
            if (options.status === "succeeded") {
                if(!options.value.targetRestUrl)
                        // ERROR: this property should be populated!!
                        console.error("No targetRestUrl was returned");
                else {
                        // we are all happy do fetch
                        fetch(options.value.targetRestUrl + "/v2.0/users/" + options.value.targetMailbox + "/messages/" + id
                    , {
                        headers: {
                            'Authorization': "Bearer " + result.value,
                            'Content-Type': 'application/json'
                        }
                    }
                ).then(function (r) {
                    console.log(r);
                }).catch(function (e) {
                    console.log("error:", e)
                });
            } else {
                console.log("Error fetching shared options");
            }
        });
    } else {
        console.log("Error obtaining token.");
    }
});

```
### NOTE: 
This is not the same as #1140  - The call to getSharedPropertiesAsync returns successfully but the targetRestUrl is absent.

While this is similar to #1161 - that ticket has been closed because _it failed to outline clearly the result of the issue_ being that there is no way to construct a valid REST URL to the delegated shared resource (message or calendar item). To spell it out the **Office.context.mailbox.item.getSharedPropertiesAsync** method returns an object that should contain targetRestUrl but doesn't.

### You Environment
Platform [PC desktop, Mac, iOS, Office Online]: PC desktop
Host [Excel, Word, PowerPoint, etc.]: Outlook WA
Office version number: Office 365 online
Operating System: Windows 10
Browser (if using Office Online): Chrome

**Comment by exextoc** [comment]
Hi @davyarris , we are unable to reproduce this on outlook.office.com. This seems to be very account-specific. Is it possible to get a test account for which you're experiencing this issue, so that we can reproduce it on our end?

**Comment by davyarris** [comment]
Hmmm, I am using my own MS account atm. I guess I could look at getting a test account set up... It might be best for me to see if others in my org are experiencing the same problem. seems unusual.

**Comment by davyarris** [comment]
I wonder does the value usually returned by this property usually just return the Outlook REST API base URL - https://outlook.office.com/, even in situations where you are accessing on-premises?

**Comment by davyarris** [comment]
Hi @exextoc 

As it turns out this was my bug. Massive apologies for wasting your time :|

Regards

Dave

**Comment by davyarris** [comment]
Re-opening issue... Seems like it is still a problem:

![image](https://user-images.githubusercontent.com/59075773/85952441-a90df880-b9ac-11ea-833b-667dcfb24a24.png)

**Comment by davyarris** [comment]
Some further curious information is that:

When accessing email from https://outlook.office365.com/ then targetRestUrl returns as expected: "https://outlook.office.com/api"
but when accessing from https://outlook.office.com/ then targetRestUrl is undefined.

**Comment by exextoc** [comment]
Hey @davyarris ,

As we requested before, is it possible for you to share a test account where we can reproduce this issue? We will take a look.

**Comment by davyarris** [comment]
Hi @exextoc 

I'll have a chat to our admin and see if I can get an account set-up for this purpose. 

It's more than just my account that is affected as we have seen similar symptoms in other tickets where the problem seemed to only occur when using Outlook from https://outlook.office.com/. I recall some mention of a call to discover either an endpoint or a schema perhaps which was failing with a 404 or 503 prior to the sharedProperties being returned without targetRestUrl (Ref: [1146](https://github.com/OfficeDev/office-js/issues/1146#issuecomment-628066776))

I'll let you know how I go.

**Comment by davyarris** [comment]
Hi @exextoc 

After some investigation, I think we may have found the problem. It appears that the issue where the targetRestUrl is not returned on the sharedProperties object occurs when the Primary inbox account (i.e. this is usually the first account that was set up under your Outlook client) does not have permissions to access domain information for the account.

 @JamesBigChange and @grumpyAL - I was hoping I might be able to get your help in confirming that in your situations where this has occurred.

For example, in my situation I set-up a personal account, followed by the domain account. This has some kind of effect on the permissions allowed by Outlook. To confirm this theory, I simply cleared all Outlook accounts and re-setup my Outlook containing only my domain account. I now get the targetRestUrl on the shared properties.

Once confirmed I hope we should be able to close this one.

Thanks

**Comment by exextoc** [comment]
Hi @davyarris we will try to verify this scenario as well. Just to confirm, you are seeing this issue on OWA (browser version of Outlook), right? Did you remove your personal account before adding the work account?

**Comment by davyarris** [comment]
Hi @exextoc 
 good question! I actually added all accounts through Native Outlook. But the problem only seems to occur in OWA (specifically https://outlook.office.com/).
No accounts were removed. Only additional ones added (i.e. 1. Added personal, 2. added work)

**Comment by exextoc** [comment]
@davyarris Could you give detailed steps on how you added a personal and work account in OWA? We are trying to reproduce the scenario but have trouble understanding what you're doing to surface the issue.

**Comment by davyarris** [comment]
Hi Exextoc

I added my account via **NATIVE** Outlook as follows

1. Open newly installed Outlook (windows native) - you could use an existing install but remove all accounts first 
2. Add a new account  (File > Add Account...). This should be an account that does not have any access to the domain like a personal email account. For me, it was my personal outlook.com account
3. Add your company account to Outlook (File > Add Account...). 
4. Now go to OWA (specifically https://outlook.office.com/) to test the add-in, logging in with your company account.

**Comment by exextoc** [comment]
Hi @davyarris , we tried the same repro steps, but still couldn't reproduce the issue - we still get a valid targetRestUrl.

What we surmise is that this issue (targetRestUrl being absent for outlook.office.com accounts) was a recent bug that we fixed. It was rolled out two weeks back or so. The issue was probably still persisting for you due to a cached value, and once you removed the accounts from Outlook it was resolved.

Since you're not the facing it anymore, we're closing this. Thanks!
